src/main.nr

```diff
@@ -167,21 +168,39 @@ contract PrivateOracle {
         let address_this = AztecAddress::new(context.this_address());
         let mut escrow_randomness = 0;
 
+ 
+
         if(payment_token.address != 0) {
+
+            let token_contract = TokenPrivateContextInterface::at(payment_token.address);
+
             // Take the payment from the user
-            escrow_randomness = TokenPrivateContextInterface::at(payment_token.address).escrow(
+            escrow_randomness = token_contract.escrow(
                 &mut context,
                 FromEscrowStruct { address: from.address },
-                AgentEscrowStruct { address: address_this.address },
+                OwnerEscrowStruct { address: address_this.address },
                 fee,
-                [
-                    ParticipantsEscrowStruct { address: from.address },
-                    ParticipantsEscrowStruct { address: divinity_address.address },
-                    ParticipantsEscrowStruct { address: 0 },
-                    ParticipantsEscrowStruct { address: 0 }
-                ],
                 nonce
             )[0];
+
+            // Broadcast the escrow to the user
+            let _ = token_contract.broadcast_escrow_note_for(
+                &mut context,
+                AccountBroadcastEscrowNoteForStruct { address: from.address },
+                OwnerBroadcastEscrowNoteForStruct { address: address_this.address },
+                fee,
+                escrow_randomness
+            );
+
+            // Broadcast the escrow to the divinity
+            let _ = token_contract.broadcast_escrow_note_for(
+                &mut context,
+                AccountBroadcastEscrowNoteForStruct { address: divinity_address.address },
+                OwnerBroadcastEscrowNoteForStruct { address: address_this.address },
+                fee,
+                escrow_randomness
+            );
```

Comment:
Related to the other comment. This way of filtering:

```rs
   // Assert if question doesn't already exist from the same requester
    let question_getter_option = NoteGetterOptions::new().select(0, question).select(1, from.address).set_limit(1);
    let question_note = storage.questions.get_notes(question_getter_option)[0];
```

Doesn't filter for `divinity_address`, which disallows a user from making the same question to the different divinities -- which I don't know it's part of the design, but it seems like a probable case. If this is changed, the filtering logic in cancel_question should be changed as well.

Another thing that came to mind is that this forbids asking the same question to the divinity twice, but what if the question was one that changed given the date. For example, a vague question like: "What's the weather in Buenos Aires?"

Lastly, divinities can't override answers. I was wondering if this made sense in the case of a fat finger given that the divinity is trusted (I assume), otherwise they could just answer silly things and increase their balances.

I guess this would add complexity, but the thought just crossed my head.

Reply:
@0xng Divinity can be trusted or not, then we need to settle it in a way or another (M/N from Skele, a proof of correct answer if applicable, etc)

Reply:
@drgorillamd Didn't follow the last part. With the current design, the divinity gets paid upon answer submission, as in its private balance increases without check on its correctness. Are you saying that a "resolution layer" will be added to ensure correctness sometime ahead?